### ISSUE TITLE
Installing Directories without the Directories DB module causes WSOD when viewing directory channel pages

### ISSUE THREAD

**Issue by stephen-cox** [localgovdrupal/localgov_directories#227]
On a fresh LocalGov install, enable localgov_directories_page (which enables localgov_directories_page, localgov_directories, search_api, search_api_db, facets), create some directory facets, and then a directory channel. The error below is displayed when trying to view the directory channel. Enabling the localgov_directories_db module fixes the issue.

Should we be enabling the localgov_directories_db module on install?

We are enabling the search_api_db. Maybe we need to remove the dependency on this from the localgov_directories.info.yml file and then add a check for Solr (or similar) and enable localgov_directories_db and dependencies if no other search backend is installed.

@ekes What are your thoughts on this?

```
The website encountered an unexpected error. Please try again later.

Drupal\Component\Plugin\Exception\PluginNotFoundException: The "search_api:views_embed__localgov_directory_channel__node_embed" plugin does not exist. Valid plugin IDs for Drupal\facets\FacetSource\FacetSourcePluginManager are: in Drupal\Core\Plugin\DefaultPluginManager->doGetDefinition() (line 53 of core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php).

Drupal\Core\Plugin\DefaultPluginManager->getDefinition('search_api:views_embed__localgov_directory_channel__node_embed') (Line: 16)
Drupal\Core\Plugin\Factory\ContainerFactory->createInstance('search_api:views_embed__localgov_directory_channel__node_embed', Array) (Line: 83)
Drupal\Component\Plugin\PluginManagerBase->createInstance('search_api:views_embed__localgov_directory_channel__node_embed') (Line: 463)
Drupal\facets\FacetManager\DefaultFacetManager->updateResults('search_api:views_embed__localgov_directory_channel__node_embed') (Line: 250)
Drupal\facets\FacetManager\DefaultFacetManager->processFacets('search_api:views_embed__localgov_directory_channel__node_embed') (Line: 327)
Drupal\facets\FacetManager\DefaultFacetManager->processBuild(Object) (Line: 387)
Drupal\facets\FacetManager\DefaultFacetManager->build(Object) (Line: 90)
Drupal\facets\Plugin\Block\FacetBlock->build() (Line: 171)
Drupal\block\BlockViewBuilder::preRender(Array)
call_user_func_array(Array, Array) (Line: 101)
Drupal\Core\Render\Renderer->doTrustedCallback(Array, Array, 'Render #pre_render callbacks must be methods of a class that implements \Drupal\Core\Security\TrustedCallbackInterface or be an anonymous function. The callback was %s. See https://www.drupal.org/node/2966725', 'exception', 'Drupal\Core\Render\Element\RenderCallbackInterface') (Line: 772)
Drupal\Core\Render\Renderer->doCallback('#pre_render', Array, Array) (Line: 363)
Drupal\Core\Render\Renderer->doRender(Array, 1) (Line: 201)
Drupal\Core\Render\Renderer->render(Array, 1) (Line: 157)
Drupal\Core\Render\Renderer->Drupal\Core\Render\{closure}() (Line: 564)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 158)
Drupal\Core\Render\Renderer->renderPlain(Array) (Line: 172)
Drupal\Core\Render\Renderer->renderPlaceholder('', Array) (Line: 649)
Drupal\Core\Render\Renderer->replacePlaceholders(Array) (Line: 534)
Drupal\Core\Render\Renderer->doRender(Array, 1) (Line: 201)
Drupal\Core\Render\Renderer->render(Array, 1) (Line: 157)
Drupal\Core\Render\Renderer->Drupal\Core\Render\{closure}() (Line: 564)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 158)
Drupal\Core\Render\Renderer->renderPlain(Array) (Line: 57)
localgov_base_preprocess_page(Array, 'page', Array)
call_user_func_array('localgov_base_preprocess_page', Array) (Line: 287)
Drupal\Core\Theme\ThemeManager->render('page', Array) (Line: 422)
Drupal\Core\Render\Renderer->doRender(Array, ) (Line: 201)
Drupal\Core\Render\Renderer->render(Array) (Line: 479)
Drupal\Core\Template\TwigExtension->escapeFilter(Object, Array, 'html', NULL, 1) (Line: 117)
__TwigTemplate_093b3e997170386224fd56a55ecbf0d5->doDisplay(Array, Array) (Line: 405)
Twig\Template->displayWithErrorHandling(Array, Array) (Line: 378)
Twig\Template->display(Array, Array) (Line: 44)
__TwigTemplate_f98e741e3186ca3a5724c336deb755a8->doDisplay(Array, Array) (Line: 405)
Twig\Template->displayWithErrorHandling(Array, Array) (Line: 378)
Twig\Template->display(Array) (Line: 390)
Twig\Template->render(Array) (Line: 55)
twig_render_template('themes/contrib/localgov_scarfolk/templates/layout/html.html.twig', Array) (Line: 384)
Drupal\Core\Theme\ThemeManager->render('html', Array) (Line: 422)
Drupal\Core\Render\Renderer->doRender(Array, ) (Line: 201)
Drupal\Core\Render\Renderer->render(Array) (Line: 162)
Drupal\Core\Render\MainContent\HtmlRenderer->Drupal\Core\Render\MainContent\{closure}() (Line: 564)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 163)
Drupal\Core\Render\MainContent\HtmlRenderer->renderResponse(Array, Object, Object) (Line: 90)
Drupal\Core\EventSubscriber\MainContentViewSubscriber->onViewRenderArray(Object, 'kernel.view', Object)
call_user_func(Array, Object, 'kernel.view', Object) (Line: 142)
Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch(Object, 'kernel.view') (Line: 164)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 81)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 58)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 709)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

**Comment by ekes** [comment]
This is a 'bug' in facets I think. When there is a disabled index Search API reports 'Cannot search on a disabled index.'; but it seems here Facets is still going ahead and trying to render.

I'll look to see if we can't:-
  (a) make sure we tell users to enable a search backend, so that the index is enabled;
  (b) fix the facets block to do the same as Search API to report a disabled index.

**Comment by ekes** [comment]
(b) My not very much tested patch https://www.drupal.org/project/facets/issues/3311856
To be more convincing I'd need to check it against all the different situations that the block can be in.

**Comment by stephen-cox** [comment]
Fixed with #233
